server/src/test/java/org/opensearch/search/aggregations/support/IncludeExcludeTests.java

```diff
@@ -297,4 +298,114 @@ private IncludeExclude serializeMixedRegex(IncludeExclude incExc) throws IOExcep
         }
     }
 
+    private static BytesRef[] toBytesRefArray(String... values) {
+        BytesRef[] bytesRefs = new BytesRef[values.length];
+        for (int i = 0; i < values.length; i++) {
+            bytesRefs[i] = new BytesRef(values[i]);
+        }
+        return bytesRefs;
+    }
+
+    public void testPrefixOrds() throws IOException {
```

Comment:
I was wondering if we could add a negative test where a certain regex will return null and fall back to the default automaton look up?

Reply:
That's kind of what we're doing with the test on line 397, but it's really limited. (We fail fast at `IncludeExclude:806`, when we check if both strings end with `.*`.)

I think I'll make `expandRegexp` package-private, so it can be tested in isolation. That's where a lot of the "fancy" logic resides anyway. (And it's the method that we might modify in future to cover more regexp cases.)